### ISSUE TITLE
Regolith starts then goes back to GDM on Ubuntu 20.04

### ISSUE THREAD

**Issue by nirvdrum** [regolith-linux/regolith-desktop#397]
**Describe the bug**

I've set Regolith up for the first time in Ubuntu 20.04 (i.e., not an upgrade) and I'm running into a problem whereby Regolith starts up fine, but then appears to crash and goes back to the GDM login screen. I see enough rendered to get the prompt to create the config files, but it crashes before I can enter anything.

**To Reproduce**

I don't really know how to reproduce this, unfortunately. I have no prior experience with Regolith, so I never had a configuration that I know works. I have fallen back to using i3 + MATE and that's working fine, on the same machine.

**Expected behavior**

I'd expect the i3 session to remain running so I can work with it.

**Screenshots**

**Installation Details**
 - Regolith Install Type: PPA
 - Regolith Version: 2.63-1focal1
 - PPA url: ppa:regolith-linux/release
 - Host OS (for PPA): Ubuntu 20.04

**Additional context**

I'm not sure that this really matters, but I'm running in a dual monitor set up. I have two GPUs installed as well, although only one is currently being used. The other exists for for VM pass-through.

Here's what my syslog looks like when starting up and crashing, grepping for "i3". I can give the full log if helpful. Searching through the logs, the issue looks similar to #141, due to the "ERROR: Another window manager seems to be running (X error 10)". While I have MATE installed, I've verified this happens on a reboot where I haven't initiated any DE sessions.

<details>
<summary>/var/log/syslog snippet</summary>

```
May 14 09:05:40 melchior i3-gnome-flashback.desktop[42084]: method return time=1589461540.323354 sender=:1.32 -> destination=:1.61 serial=227 reply_serial=2
May 14 09:05:40 melchior i3-gnome-flashback.desktop[42084]:    object path "/org/gnome/SessionManager/Client23"
May 14 09:05:40 melchior i3-gnome-flashback.desktop[42103]: md5sum: /home/nirvdrum/.Xresources-regolith: No such file or directory
May 14 09:05:40 melchior i3.desktop[42329]: /bin/sh: 1: gitter: not found
May 14 09:05:41 melchior i3.desktop[42294]: [{"success":true},{"success":true}]
May 14 09:05:41 melchior i3-gnome-flashback.desktop[42077]: Regolith is launching i3-gaps with /etc/regolith/i3/config
May 14 09:05:41 melchior i3.desktop[42498]: ERROR block_reap:328: [bandwidth] bad exit code 1
May 14 09:05:41 melchior i3-gnome-flashback.desktop[42573]: 05/14/2020 09:05:41 AM - ERROR: Another window manager seems to be running (X error 10)
May 14 09:05:41 melchior i3-gnome-flashback.desktop[42754]: method return time=1589461541.666493 sender=:1.32 -> destination=:1.74 serial=273 reply_serial=2
May 14 09:05:42 melchior i3.desktop[42362]: [../../i3-gaps-wm-4.17.1/i3bar/src/xcb.c:1078] ERROR: X11 connection was closed unexpectedly - maybe your X server terminated / crashed?
May 14 09:05:42 melchior i3.desktop[42362]: [libi3] ../../i3-gaps-wm-4.17.1/libi3/font.c Using Pango font System San Francisco Display, size 14
May 14 09:05:42 melchior i3.desktop[42337]: Pidgin: Fatal IO error 2 (No such file or directory) on X server :0.
May 14 09:05:43 melchior i3.desktop[42176]: i3: Cannot open display
May 14 09:05:43 melchior i3.desktop[42353]: [42318:42318:0514/090543.525272:ERROR:browser_main_loop.cc(1485)] Unable to open X display.
May 14 09:05:37 melchior /usr/lib/gdm3/gdm-x-session[37305]: (II) Loading sub module "dri3"
May 14 09:05:37 melchior /usr/lib/gdm3/gdm-x-session[37305]: (II) LoadModule: "dri3"
May 14 09:05:37 melchior /usr/lib/gdm3/gdm-x-session[37305]: (II) Module "dri3" already built-in
May 14 09:05:40 melchior i3-gnome-flashback.desktop[42084]: method return time=1589461540.323354 sender=:1.32 -> destination=:1.61 serial=227 reply_serial=2
May 14 09:05:40 melchior i3-gnome-flashback.desktop[42084]:    object path "/org/gnome/SessionManager/Client23"
May 14 09:05:40 melchior i3-gnome-flashback.desktop[42103]: md5sum: /home/nirvdrum/.Xresources-regolith: No such file or directory
May 14 09:05:40 melchior i3.desktop[42329]: /bin/sh: 1: gitter: not found
May 14 09:05:41 melchior i3.desktop[42294]: [{"success":true},{"success":true}]
May 14 09:05:41 melchior i3-gnome-flashback.desktop[42077]: Regolith is launching i3-gaps with /etc/regolith/i3/config
May 14 09:05:41 melchior i3.desktop[42498]: ERROR block_reap:328: [bandwidth] bad exit code 1
May 14 09:05:41 melchior i3-gnome-flashback.desktop[42573]: 05/14/2020 09:05:41 AM - ERROR: Another window manager seems to be running (X error 10)
May 14 09:05:41 melchior i3-gnome-flashback.desktop[42754]: method return time=1589461541.666493 sender=:1.32 -> destination=:1.74 serial=273 reply_serial=2
May 14 09:05:42 melchior i3.desktop[42362]: [../../i3-gaps-wm-4.17.1/i3bar/src/xcb.c:1078] ERROR: X11 connection was closed unexpectedly - maybe your X server terminated / crashed?
May 14 09:05:42 melchior i3.desktop[42362]: [libi3] ../../i3-gaps-wm-4.17.1/libi3/font.c Using Pango font System San Francisco Display, size 14
May 14 09:05:42 melchior i3.desktop[42337]: Pidgin: Fatal IO error 2 (No such file or directory) on X server :0.
May 14 09:05:43 melchior i3.desktop[42176]: i3: Cannot open display
May 14 09:05:43 melchior i3.desktop[42353]: [42318:42318:0514/090543.525272:ERROR:browser_main_loop.cc(1485)] Unable to open X display.
```
</details>

**Comment by kgilmer** [comment]
Hi @nirvdrum :smile: .  Thanks for the detailed information.  Based on the log data you provided, something is starting a window manager on your system before Regolith's instance of i3 has a chance to start.  I don't have any experience with MATE compatibility, so cannot say either way if it's related.  I also don't have any specific steps to recommend to troubleshoot, other than perhaps change `/usr/bin/i3-gnome-flashback` to print active processes before launching i3 `ps aux`, and then try and find out what window manager is running and then try and trace back to whatever is starting it.  You may also want to try switching to the lightdm display manager to see if that makes a difference.

**Comment by nirvdrum** [comment]
Thanks for the reply. I've tried to work it out for the past couple hours, but it's turning into a game of whack-a-mole. I keep uninstalling packages and rebooting the system for good measure, Regolith runs for a few minutes and then crashes due to something else. Once it crashes, any subsequent login will crash immediately. 

I'm at a bit of a loss as to what it is, since I've been running Ubuntu's provided i3-wm, i3's packaged i3, and manually built i3-gaps, running both within and without of Mate for several years and have never encountered this problem. I don't think I'm running anything that's fundamentally incompatible with i3. The only thing I can really see consistently in the logs around the crash is Evolution, Snap updates, Keybase (electron app), and some VMware code for the virtual display driver, all of which work with my non-Regolith i3 environment.

It sounds like Regolith needs to be the only window manager installed. Or at least it sounds like the issue I've encountered is somewhat expected, so I'm happy to close this out as an environmental issue. I had been hoping Regolith might provide a nicer package than what I'm using, but I'd need to be able to have time to transition over to it, which appears to be a conflicting requirement. Thanks again for taking the time to review my issue.
